File: 01_Objectif-C/14_protocole/14_protocole/ViewController.m
//
//  ViewController.m
//  14_protocole
//
//  Created by etudiant on 4/18/17.
//  Copyright © 2017 etudiant. All rights reserved.
//

#import "ViewController.h"
#import "Personne.h"
#import "Employe.h"
#import "Achat.h"
@interface ViewController ()

@end

@implementation ViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
   NSMutableArray * registre = [NSMutableArray new];
    Personne * p = [Personne new];
    p.nom = @"Marie";
    
   NSLog(@"%@", p);
    [registre addObject:p];
    
    Employe* e = [[Employe alloc] initAvecNom:@"Pacifique" etSalaire:3000];
    
   [registre addObject:e];
    
 Achat * a = [[Achat alloc] initAvecPrixHT:100 etTypeTVA:@"tva20"];
   [registre addObject:a];/**/
    

   float recette = 0;
    float impotItem=0;
    for(NSObject* item in registre)
    {
        if([item conformsToProtocol:@protocol(PImposable)])
        {
            impotItem =((id<PImposable>)item).calculerImpot;
            NSLog(@"Type impot:%@, montant: %f", ((id<PImposable>)item).nomImpot,impotItem);
                  
                  recette+= impotItem;
        }
    }
    NSLog(@"La recette totale du fisc: %0.2f",recette);
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 03_Swift_fondamentaux/Explorer/Explorer/ViewController.swift
//
//  ViewController.swift
//  Explorer
//
//  Created by etudiant on 10/05/2017.
//  Copyright © 2017 -. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDataSource
{

    let favouriteThing = ["Cordova","Linux"]
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.favouriteThing.count;
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "MyList")
        cell?.textLabel?.text = self.favouriteThing[indexPath.row]
        
        return cell!
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

